platforms/jetson/jetson_adaptor.go

```diff
@@ -0,0 +1,291 @@
+package jetson
+
+import (
+	"errors"
+	"fmt"
+
+	"sync"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"gobot.io/x/gobot"
+	"gobot.io/x/gobot/drivers/i2c"
+	"gobot.io/x/gobot/drivers/spi"
+	"gobot.io/x/gobot/sysfs"
+)
+
+// Adaptor is the Gobot Adaptor for the Jetson Nano
+type Adaptor struct {
+	mutex              *sync.Mutex
+	name               string
+	revision           string
+	digitalPins        map[int]*sysfs.DigitalPin
+	pwmPins            map[int]*PWMPin
+	i2cDefaultBus      int
+	i2cBuses           [2]i2c.I2cDevice
+	spiDefaultBus      int
+	spiDefaultChip     int
+	spiDevices         [2]spi.Connection
+	spiDefaultMode     int
+	spiDefaultMaxSpeed int64
+	JSBlasterPeriod    uint32
+}
+
+// NewAdaptor creates a Raspi Adaptor
+func NewAdaptor() *Adaptor {
+	j := &Adaptor{
+		mutex:           &sync.Mutex{},
+		name:            gobot.DefaultName("JetsonNano"),
+		digitalPins:     make(map[int]*sysfs.DigitalPin),
+		pwmPins:         make(map[int]*PWMPin),
+		JSBlasterPeriod: 3000000,
+	}
+
+	j.i2cDefaultBus = 1
+	j.spiDefaultBus = 0
+	j.spiDefaultChip = 0
+	j.spiDefaultMode = 0
+	j.spiDefaultMaxSpeed = 10000000
+
+	return j
+}
+
+// Name returns the Adaptor's name
+func (j *Adaptor) Name() string {
+	j.mutex.Lock()
+	defer j.mutex.Unlock()
+
+	return j.name
+}
+
+// SetName sets the Adaptor's name
+func (j *Adaptor) SetName(n string) {
+	j.mutex.Lock()
+	defer j.mutex.Unlock()
+
+	j.name = n
+}
+
+// Connect starts connection with board and creates
+// digitalPins and pwmPins adaptor maps
+func (j *Adaptor) Connect() (err error) {
+	return
+}
+
+// Finalize closes connection to board and pins
+func (j *Adaptor) Finalize() (err error) {
+	j.mutex.Lock()
+	defer j.mutex.Unlock()
+
+	for _, pin := range j.digitalPins {
+		if pin != nil {
+			if perr := pin.Unexport(); err != nil {
+				err = multierror.Append(err, perr)
+			}
+		}
+	}
+
+	for _, pin := range j.pwmPins {
+		if pin != nil {
+			if perr := pin.Unexport(); err != nil {
+				err = multierror.Append(err, perr)
+			}
+		}
+	}
+
+	for _, bus := range j.i2cBuses {
+		if bus != nil {
+			if e := bus.Close(); e != nil {
+				err = multierror.Append(err, e)
+			}
+		}
+	}
+	for _, dev := range j.spiDevices {
+		if dev != nil {
+			if e := dev.Close(); e != nil {
+				err = multierror.Append(err, e)
+			}
+		}
+	}
+	return
+}
+
+// DigitalPin returns matched digitalPin for specified values
+func (j *Adaptor) DigitalPin(pin string, dir string) (sysfsPin sysfs.DigitalPinner, err error) {
+	i, err := j.translatePin(pin)
+
+	if err != nil {
+		return
+	}
+
+	currentPin, err := j.getExportedDigitalPin(i, dir)
+
+	if err != nil {
+		return
+	}
+
+	if err = currentPin.Direction(dir); err != nil {
+		return
+	}
+
+	return currentPin, nil
+}
+
+func (j *Adaptor) getExportedDigitalPin(translatedPin int, dir string) (sysfsPin sysfs.DigitalPinner, err error) {
```

Comment:
please move the local functions to the end of the file

Reply:
fix it commit https://github.com/hybridgroup/gobot/commit/ee0bff582a684a8356c8593170a79d4bec280039.